### PR TITLE
#2 - Added a register page & made some small fixes on the login page

### LINKED ISSUE #2: Register Page
Currently our register page does nothing, make it work.

**Accepttest**
* It should be possible to create a user from the register page.
* All links should work
* Redirect to login after successful creation

### PR DESCRIPTION
**Pull request by frederikpyt**
## Description
- Added functionality to the register page
- Made some minor fixes to the login page

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/components/FormSignUp.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+				<div>
+					<button
+						type="submit"
+						class="w-full rounded-lg bg-primary-700 px-5 py-3 text-center text-base font-medium text-white hover:bg-primary-800 focus:ring-4 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800 sm:w-auto"
+					>
+						Create account
+					</button>
+					<span v-if="errorMessage" class="mb-4 ml-2 w-full text-xs text-red-600">{{
+						errorMessage
+					}}</span>
```
Comment (kristianbinau) [inline src/components/FormSignUp.vue]: Hahaha, most pretty error message : ) Maybe we could use the alert from [Flowbite](https://flowbite.com/docs/components/alerts/#alerts-with-icon)

```suggestion
					<div
						v-if="errorMessage"
						class="mt-6 flex items-center rounded-lg bg-red-50 p-4 text-sm text-red-800 dark:bg-gray-700 dark:text-red-400"
						role="alert"
					>
						<svg
							class="mr-3 inline h-4 w-4 flex-shrink-0"
							aria-hidden="true"
							xmlns="http://www.w3.org/2000/svg"
							fill="currentColor"
							viewBox="0 0 20 20"
						>
							<path
								d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5ZM9.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3ZM12 15H8a1 1 0 0 1 0-2h1v-3H8a1 1 0 0 1 0-2h2a1 1 0 0 1 1 1v4h1a1 1 0 0 1 0 2Z"
							/>
						</svg>
						<span class="sr-only">Info</span>
						<div><span class="font-medium">Failure!</span> {{ errorMessage }}</div>
					</div>
```

File: src/components/FormSignUp.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	email: 'test@test.com',
+	password: 'test@test.com',
+	confirmPassword: 'test@test.com',
+});
+
+const errorMessage = ref<null | string>(null);
+
+function onSubmit() {
+	console.log(formFields.value);
+
```
Comment (kristianbinau) [inline src/components/FormSignUp.vue]:
```suggestion
```

File: src/components/FormSignUp.vue [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+const form = ref<HTMLFormElement>();
+const password = ref<HTMLInputElement>();
+const confirmPassword = ref<HTMLInputElement>();
+
+const formFields = ref({
+	name: 'John Doe',
+	email: 'test@test.com',
+	password: 'test@test.com',
+	confirmPassword: 'test@test.com',
```
Comment (kristianbinau) [inline src/components/FormSignUp.vue]: Let's take this opportunity to remove the default values.
```suggestion
	name: '',
	email: '',
	password: '',
	confirmPassword: ',
```

File: src/components/FormSignIn.vue [hunk 4]
```diff
@@ -72,14 +77,19 @@ import { asset, api, url } from '@lib/helpers';
 import { SITE_TITLE_SHORT } from '@lib/constants';
 import { ref } from 'vue';
 
+const form = ref<HTMLFormElement>();
 const formFields = ref({
 	email: 'test@test.com',
 	password: 'test@test.com',
```
Comment (kristianbinau) [inline src/components/FormSignIn.vue]: Let's take this opportunity to remove the default values.
```suggestion
	email: '',
	password: '',
```

File: src/components/FormSignIn.vue [hunk 5]
```diff
@@ -72,14 +77,19 @@ import { asset, api, url } from '@lib/helpers';
...
 });
 
+const errorMessage = ref<null | string>(null);
+
 function onSubmit() {
 	console.log(formFields.value);
```
Comment (kristianbinau) [inline src/components/FormSignIn.vue]:
```suggestion
```

File: src/components/FormSignIn.vue [hunk 6]
```diff
@@ -48,12 +48,17 @@
...
+				<div>
+					<button
+						type="submit"
+						class="w-full rounded-lg bg-primary-700 px-5 py-3 text-center text-base font-medium text-white hover:bg-primary-800 focus:ring-4 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800 sm:w-auto"
+					>
+						Login to your account
+					</button>
+					<span v-if="errorMessage" class="mb-4 ml-2 w-full text-xs text-red-600">{{
+						errorMessage
+					}}</span>
```
Comment (kristianbinau) [inline src/components/FormSignIn.vue]: Hahaha, most pretty error message : ) Maybe we could use the alert from [Flowbite](https://flowbite.com/docs/components/alerts/#alerts-with-icon)

```suggestion
					<div
						v-if="errorMessage"
						class="mt-6 flex items-center rounded-lg bg-red-50 p-4 text-sm text-red-800 dark:bg-gray-700 dark:text-red-400"
						role="alert"
					>
						<svg
							class="mr-3 inline h-4 w-4 flex-shrink-0"
							aria-hidden="true"
							xmlns="http://www.w3.org/2000/svg"
							fill="currentColor"
							viewBox="0 0 20 20"
						>
							<path
								d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5ZM9.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3ZM12 15H8a1 1 0 0 1 0-2h1v-3H8a1 1 0 0 1 0-2h2a1 1 0 0 1 1 1v4h1a1 1 0 0 1 0 2Z"
							/>
						</svg>
						<span class="sr-only">Info</span>
						<div><span class="font-medium">Failure!</span> {{ errorMessage }}</div>
					</div>
```

File: src/components/FormSignUp.vue [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+					/>
+				</div>
+				<div>
+					<label for="password" class="mb-2 block text-sm font-medium text-gray-900 dark:text-white"
+						>Your password</label
+					>
+					<input
+						ref="password"
+						v-model="formFields.password"
+						type="password"
```
Comment (kristianbinau) [inline src/components/FormSignUp.vue]: Should we have a show hide password thing? Would be cool if we could add it as a directive.

File: src/components/FormSignUp.vue [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+				}
+
+				throw new Error(await response.text());
+			}
+
+			return response.json();
+		})
+		.then((data) => {
+			console.log('Success:', data);
+			window.location.href = url('auth/login');
```
Comment (kristianbinau) [inline src/components/FormSignUp.vue:124]: Would be cool if we added a [Success alert](https://flowbite.com/docs/components/alerts/#alerts-with-icon) under the submit button and waited a few seconds before redirect.

I think that would make the most UX sense.

**Review by kristianbinau — CHANGES_REQUESTED** [review]

File: src/components/FormSignUp.vue (hunk 8 above)
Comment (frederikpyt) [reply src/components/FormSignUp.vue:124]: Google and other big sites don't have a wait before redirect?

Comment (kristianbinau) [reply src/components/FormSignUp.vue:124]: I guess that's true, you're off the hook then

File: src/components/base/button.vue [hunk 9]
```diff
@@ -0,0 +1,34 @@
+<template>
+	<button
+		:type="props.type"
+		:class="[
+			props.color === 'primary'
+				? 'bg-primary-700 text-white hover:bg-primary-800 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800'
+				: '',
+			props.color === 'secondary'
+				? 'bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-100 dark:bg-primary-400 dark:hover:bg-primary-500 dark:focus:ring-primary-600'
+				: '',
```
Comment (kristianbinau) [inline src/components/base/button.vue]: I find this a bit ugly, maybe we could move it into computed properties. Like this maybe?
```js
const buttonClasses = computed(() => {
	if (props.color === 'primary') {
		return 'bg-primary-700 text-white hover:bg-primary-800 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800';
	} else if (props.color === 'secondary') {
		return 'bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-100 dark:bg-primary-400 dark:hover:bg-primary-500 dark:focus:ring-primary-600';
	} else {
		return '';
	}
})
```

File: src/components/base/button.vue [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+<script setup lang="ts">
+import type { PropType } from 'vue';
+
+const props = defineProps({
+	type: {
+		type: String as PropType<'submit' | 'reset' | 'button'>,
+		required: true,
+	},
+	color: {
+		type: String as PropType<'primary' | 'secondary'>,
```
Comment (kristianbinau) [inline src/components/base/button.vue]: Would this be a good time for Enums? Then we could also remove the "magic strings" that we use around the component? Making it more type-safe, less likely to type a string wrong.

Like this?

```vue
<template>
	<button
		:type="props.type"
		:class="[
			props.color === ButtonColor.Primary
				? 'bg-primary-700 text-white hover:bg-primary-800 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800'
				: '',
			props.color === ButtonColor.Secondary
				? 'bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-100 dark:bg-primary-400 dark:hover:bg-primary-500 dark:focus:ring-primary-600'
				: '',
		]"
		class="w-full rounded-lg px-5 py-3 text-center text-base font-medium focus:ring-4 sm:w-auto"
		@click="emit('click')"
	>
		<slot />
	</button>
</template>

<script lang="ts">
export enum ButtonColor {
	Primary = 'primary',
	Secondary = 'secondary',
}
</script>

<script setup lang="ts">
import type { PropType } from 'vue';

const props = defineProps({
	type: {
		type: String as PropType<'submit' | 'reset' | 'button'>,
		required: true,
	},
	color: {
		type: String as PropType<ButtonColor>,
		required: true,
	},
});

const emit = defineEmits(['click']);
</script>
```
image

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Base components look awesome!

**Review by kristianbinau — APPROVED** [review]
I'm in love!

**Pull request merged**